lib/ansible/modules/cloud/docker/docker_container.py

```diff
@@ -1488,6 +1547,33 @@ def _get_network_id(self, network_name):
             self.fail("Error getting network id for %s - %s" % (network_name, str(exc)))
         return network_id
 
+    def _process_rate_bps(self, option):
+        if not isinstance(getattr(self, option), list):
+            self.fail("'{0}' expected to be list of devices".format(option))
+
+        devices_list = []
+        for v in getattr(self, option):
+            device_dict = dict((x.title(), y) for x, y in v.items())
+            device_dict['Rate'] = human_to_bytes(device_dict.get('Rate', 0).upper())
```

Comment:
What happens if `'Rate' not in device_dict`? Then `0.upper()` is called, which will fail. I think you can simply remove the `upper()`; then it will work. (`human_to_bytes()` converts its argument to a string anyway, so `0` is not a problem.)

Reply:
(Also, `human_to_bytes()` handles the case as well, units will be converted to upper case.)